src/Quahog/Client.php

```diff
@@ -257,14 +261,34 @@ private function _sendCommand($command)
     /**
      * A wrapper to cleanly read a response from clamd.
      *
+     * @param bool $removeId
+     * @param string $readUntil
      * @return string
+     * @throws ConnectionException
      */
-    private function _receiveResponse()
+    private function _receiveResponse($removeId = false, $readUntil = "\n")
     {
-        $result = $this->_socket->read(4096);
+        $result = null;
+        $readUntilLen = strlen($readUntil);
+        do {
+            if ($this->_socket->selectRead($this->_timeout)) {
+                $rt = $this->_socket->read(4096, PHP_NORMAL_READ);
+                if ($rt == "") {
+                    break;
+                }
+                $result .= $rt;
+                if (strcmp(substr($result, 0 - $readUntilLen), $readUntil) == 0) {
+                    break;
+                }
+            } else {
+                throw new ConnectionException("Timeout waiting to read response");
```

Comment:
This seems to introduce a BC break. The exception handling in `_closeConnection()` is kind of weird because an exception can't be thrown there, so yours would be the introduction of an exception for many of the class' public methods. Depending on how complex the users' code are who are using this lib, this can change the behaviour of many applications.

@jonjomckay What's your opinion on that?